scripts/source-maintenance.sh

```diff
@@ -401,16 +447,26 @@ printAmFile TRANSLATE_LANGUAGES "doc/manuals/" "*.po" | sed 's%\.po%\.lang%g' >
 printAmFile STUB_SOURCE "src/" "tests/stub_*.cc" > src/tests/Stub.am
 
 # Build the GPERF generated content
-make -C src/http gperf-files
+if [ -f src/http/Makefile ]; then
+    make -C src/http gperf-files
+else
+    echo "source tree not configured; skipping regeneration of xperf sources"
+    echo "to run it, run  ./bootstrap.sh && ./configure && $0"
+fi
 
 run_ checkMakeNamedErrorDetails || exit 1
 
 # Run formatting
 echo "" >doc/debug-sections.tmp
 srcFormat || exit 1
-sort -u <doc/debug-sections.tmp | sort -n >doc/debug-sections.tmp2
-cat scripts/boilerplate.h doc/debug-sections.tmp2 >doc/debug-sections.txt
-rm doc/debug-sections.tmp doc/debug-sections.tmp2
-rm -f boilerplate_fix.sed
+
+if [ -z "$OnlyChangedSince" ]; then
+    sort -u <doc/debug-sections.tmp | sort -n >doc/debug-sections.tmp2
+    cat scripts/boilerplate.h doc/debug-sections.tmp2 >doc/debug-sections.txt
+    rm doc/debug-sections.tmp doc/debug-sections.tmp2
+    rm -f boilerplate_fix.sed
+else
+    echo "--only-changed-since specified, Skipping update of doc/debug-sections.txt"
+fi
```

Comment:
Only the tmp2 file is created within this scope, so it is the only one that should be cleaned up here. The others should remain at the top level scope where they are built.

```suggestion
    rm doc/debug-sections.tmp2
else
    echo "--only-changed-since specified, Skipping update of doc/debug-sections.txt"
fi
rm doc/debug-sections.tmp doc/debug-sections.tmp2
rm -f boilerplate_fix.sed
```